bayes_kit/drghmc.py

```diff
@@ -0,0 +1,437 @@
+from typing import Iterator, Optional, Union, Callable
+
+import numpy as np
+
+from .typing import DrawAndLogP, GradModel, Seed, VectorType
+
+
+class DrGhmcDiag:
+    """Generalized HMC sampler with (Probabilistic) Delayed Rejection diagonal metric.
+
+    To sample from a target distribution, DrGhmcDiag proposes a new sample from the
+    current sample using the leapfrog integrator. The probability of accepting this
```

Comment:
Because I am mentioning Chirag's/your paper, I am going with option (b). Thus I am removing the line about leapfrog integration. Are there any other basic HMC details that should be removed?